File: lectures/week5_lecture/async-examples/01-futures-basics.js
async function myFunc() {
	console.log("\tBefore delay");
	await new Promise(resolved => setTimeout(resolved, 500));
	console.log("\tAfter delay");
}

async function main() {
	console.log("Before call");
	var prom = myFunc();
	console.log("After call");
	await prom;
	console.log("After await");
}

main()

File: lectures/week5_lecture/async-examples/01-futures-basics.dart
Future<void> myFunc() async {
    print("\tBefore delay");
    await Future.delayed(Duration(milliseconds: 500));
    print("\tAfter delay");
}

Future<void> main() async {
    print("Before call");
    var prom = myFunc();
    print("After call");
    await prom;
    print("After await");
}
